src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs

```diff
@@ -20,6 +20,9 @@ public static class HttpClientFactoryServiceCollectionExtensions
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/>.</param>
         /// <returns>The <see cref="IServiceCollection"/>.</returns>
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "ServiceDescriptor has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```

Comment:
This one looks like it silences warnings from ServiceDescriptor static methods, and also from TryAddTransient/TryAddSingleton - I wonder if that's worth mentioning in the justification.
```suggestion
            Justification = "ServiceDescriptor, TryAddTransient, and TryAddSingleton have RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```